test/test_utils.py

```diff
@@ -361,6 +361,77 @@ def test_draw_keypoints_colored(colors):
     assert_equal(img, img_cp)
 
 
+@pytest.mark.parametrize("connectivity", [[(0, 1)], [(0, 1), (1, 2)]])
+@pytest.mark.parametrize(
+    "vis",
+    [
+        torch.tensor([[1, 1, 0], [1, 1, 0]], dtype=torch.bool),
+        torch.tensor([[1, 1, 0], [1, 1, 0]], dtype=torch.float).unsqueeze_(-1),
+    ],
+)
+def test_draw_keypoints_visibility(connectivity, vis):
+    # Keypoints is declared on top as global variable
+    keypoints_cp = keypoints.clone()
+
+    img = torch.full((3, 100, 100), 0, dtype=torch.uint8)
+    img_cp = img.clone()
+
+    vis_cp = vis if vis is None else vis.clone()
+
+    result = utils.draw_keypoints(
+        image=img,
+        keypoints=keypoints,
+        connectivity=connectivity,
+        colors="red",
+        visibility=vis,
+    )
+    assert result.size(0) == 3
+    assert_equal(keypoints, keypoints_cp)
+    assert_equal(img, img_cp)
+
+    # compare with a fakedata image
+    # connect the key points 0 to 1 for both skeletons and do not show the other key points
+    path = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), "assets", "fakedata", "draw_keypoints_visibility.png"
+    )
+    if not os.path.exists(path):
+        res = Image.fromarray(result.permute(1, 2, 0).contiguous().numpy())
+        res.save(path)
+
+    expected = torch.as_tensor(np.array(Image.open(path))).permute(2, 0, 1)
+    assert_equal(result, expected)
```

Comment:
Thank you for adding this test.

I think we'll have to `git add` the image and commit it as part of this PR so that it's within the repo. The reason being that our CI jobs start from a fresh VM so if the image wasn't present, the job would create it and compare the output against itself, which would make this test mostly useless.

(I think this `if not os.path.exists(path):` logic is mostly useless, but I realize it doesn't come from this PR so it's OK to keep it here).

Reply:
Just keep it, it *may* exist because of some internal CI stuff. We also run the torchvision tests on the Meta-internal CI and maybe there was an issue with those files internally... not sure. I'll open an issue to track this and address all existing cases together

Reply:
In the last commit I removed all the exists checks from this test suite. If you dont want that, it is all contained in that one commit. Everythin else is separate.

Reply:
I reverted the commit, now everything should be as expected.